rioxarray/rioxarray.py

```diff
@@ -1051,8 +1051,81 @@ def slice_xy(self, minx, miny, maxx, maxy):
         subset.attrs["transform"] = tuple(self.transform(recalc=True))
         return subset
 
-    def clip_box(self, minx, miny, maxx, maxy, auto_expand=False, auto_expand_limit=3):
-        """Clip the :class:`xarray.DataArray` by a bounding box.
+    def pad_xy(self, minx, miny, maxx, maxy, resolution_x, resolution_y):
```

Comment:
I am wondering if resolution_x and resolution_y need to be required here. My thoughts are that they could be pulled in from the dataset the same way it was done in `clip_box`.

Reply:
:+1: